### PR TITLE
Fix crash when evaluating background-only images

### PR DESCRIPTION
**Pull request by developer0hye**
Fix crash when evaluating background-only images

command: `python train.py -c configs/dfine/dfine_hgnetv2_n_coco.yml --test-only -r dfine_n_coco.pth`

# Before 

image

# After

image

Removed unnecessary conversion:

```python
torch.tensor(target["boxes"].tolist()).to(device),
```
since this conversion is already performed before the iteration.

```python
for samples, targets in metric_logger.log_every(data_loader, 10, header):
        samples = samples.to(device)
        targets = [{k: v.to(device) if isinstance(v, torch.Tensor) else v for k, v in t.items()} for t in targets]
```

### CODE REVIEW

**Comment by developer0hye** [comment]
It appears this issue might be related to @ArgoHA ’s PR #177. Additionally, there seems to be a potential issue with the metric values, as they appear unusually low when I checked it with coco dataset and official d-fine nano weights. 

@ArgoHA, could you please look into this? Thanks!

Metrics: {'f1': 0.00081429540827867, 'precision': 0.0009006004002668446, 'recall': 0.0007430851795789184, 'iou': np.float64(0.0003388939322393469), 'TPs': 27, 'FPs': 29953, 'FNs': 36308}

**Comment by ArgoHA** [comment]
I believe this PR fix is correct. 
On the metrics issue, I quickly looked into it, I see model predictions' labels are off, they don't overlap with ground truth labels. There are values higher than 80, so I assume this issue is happening because of `remap_mscoco_category`. I temporary "fixed" it with removing `if self.remap_mscoco_category:` in postprocessor.py, check out metrics:

Screenshot 2025-03-08 at 13 17 14

 I can't get my hands on it today to figure out the right way to fix it, but will get to it.

**Comment by Peterande** [comment]
image
https://github.com/Peterande/D-FINE/blob/7eb139b9bb433d0716a8ba45c4e7844c1d0ac90e/src/solver/det_engine.py#L207
Mapping category back to label solves the problem

**Comment by ArgoHA** [comment]
@Peterande it seems that you are remapping labels for all datasets, not only COCO. In postprocessor you have `if self.remap_mscoco_category`, it looks like we should remap only when that condition is true?

**Comment by Peterande** [comment]
> @Peterande it seems that you are remapping labels for all datasets, not only COCO. In postprocessor you have `if self.remap_mscoco_category`, it looks like we should remap only when that condition is true?

https://github.com/Peterande/D-FINE/blob/f5315882328b2ac374060dfa1d3c4548c22be543/src/solver/det_engine.py#L206-L210
You're right. Since the machines at the internship company cannot access the D-FINE environment, I manually added some corrections using the webpage edit function. If it's convenient for you, could you please check whether they are correct? Thank you!
